pages/participate/index.js

```diff
@@ -12,26 +12,44 @@ import {
   Chip,
   Tooltip,
 } from '@mui/material'
+import EventBusyIcon from '@mui/icons-material/EventBusy'
 import ISPOCard, {IspoDetail} from '../../components/ISPOCard'
 import * as fcl from '@onflow/fcl'
 import delegateToISPO from '../../cadence/web/transactions/client/delegateToISPO.cdc'
 import {toUFixString, formatCompactAmount} from '../../helpers/utils'
 import useCurrentUser from '../../hooks/useCurrentUser'
 import CardGrid from '../../layouts/CardGrid'
-import { useCurrentEpoch } from '../../hooks/epochs'
+import {useCurrentEpoch} from '../../hooks/epochs'
 
 export default function ParticipateIspoPage() {
   const {addr} = useCurrentUser()
   const res = useAccountIspos(addr)
 
   const ispos = useIspos()
+  const currentEpoch = useCurrentEpoch()
+
+  const canJoin = (epochEnd) => Number(currentEpoch) < Number(epochEnd)
 
+  // sort by time of creation
+  const sortedIsposByDate =
+    ispos &&
+    [...ispos].sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime())
+  // sort ended at the and
+  const sortedIspos =
+    sortedIsposByDate &&
+    [...sortedIsposByDate].sort((a, b) =>
+      canJoin(a.epochEnd) && !canJoin(b.epochEnd) ? -1 : 1,
```

Comment:
please someone revisit that one, I am not very familiar with `sort` fn.

Reply:
seems legit, maybe just worth pointing out that this relies on the sort function being stable (i.e. preserving the original order of items deemed equal by the comparator function), which it fortunately is by specs, since ECMA2019: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#sort_stability